airflow/dag_processing/manager.py

```diff
@@ -647,10 +650,9 @@ def set_file_paths(self, new_file_paths: list[DagFileInfo]):
         self._file_path_queue = deque(x for x in self._file_path_queue if x in new_file_paths)
         Stats.gauge("dag_processing.file_path_queue_size", len(self._file_path_queue))
 
-        # TODO: AIP-66 make callbacks bundle aware
-        # callback_paths_to_del = [x for x in self._callback_to_execute if x not in new_file_paths]
-        # for path_to_del in callback_paths_to_del:
-        #     del self._callback_to_execute[path_to_del]
+        callback_paths_to_del = [x for x in self._callback_to_execute if x not in new_file_paths]
```

Comment:
We will need to be a little smarter about this - the file could still exist in the old bundle version, but not the latest.

Reply:
So, this code was added cause of SLA callbacks: https://github.com/apache/airflow/pull/30076.  I also removed filtering of _file_path_queue in the method since there's a possibility we are just working in a different version of the bundle